### DISCUSSION TITLE
Extend type support in non-federated types

### DISCUSSION THREAD

**Discussion by SirensOfTitan** [strawberry-graphql/strawberry#2522] [General]
One fairly nice feature in the GraphQL spec is the capacity to extend types.  This is particularly worthwhile when working with `Mutation`, as you can co-locate particular mutations with the rest of their GraphQL definitions.  I've noticed the `extends` field sprinkled around in tests for federated features here, but I was wondering if there is or any support is planned for this feature.

For example, I could have a stub file, mutation.py:
```python
@strawberry.type
class Mutation:
  pass
```

... and in say, foo.py:
```python
@strawberry.type
class Foo:
  id: str
  
@strawberry.type(extends=True)
class Query:
  @strawberry.field
  def foo(id: strawberry.ID) -> Foo:
    ...
    
@strawberry.type(extends=True, name="Mutation")
class FooMutation:
  @strawberry.field
  def updateFoo(...) -> FooPayload: 
    ...
```

**Comment by patrick91** [comment]
Hi @SirensOfTitan! Thanks for the suggestion! 😊

Would you need to import FooMutation in your schema? At the moment all types have to be explicitly declared on the schema, either by using them as types of fields or by passing them to the `types` argument on the schema. 

I don't think we might allow Strawberry to automatically find types[1], as we support creating multiple schemas in a single project. 

If we don't allow finding automatic types, then maybe allowing to use extend to extend types, might not be the best DX? What do you think?

[1] but this could be something done by a framework/helper built on top of Strawberry maybe :D

**Comment by alexey-pelykh** [comment]
`@strawberry.type` has `extend` argument that seems should've solved that yet unless I'm mistaken that would still be considered a duplicate type by `validate_same_type_definition`?

**Comment by ThomasReuss** [comment]
Hi, Is there a change on this since the previous answer in 2023?
I am attempting to migrate a code base from ariadne to strawberry. I am having a large library where there is the following code in the base schema:

`union TaskResult`

and then in each of the apps, for example,

`extend union TaskResult = SpecialTaskResult`.

The advantage of this approach is that the original app only needs to know about `TaskResult` and each extension app only needs to know about their own extension. 

Can this be achieved in Strawberry by now?

**Reply by obrazkowv** [reply]
Hi, could you find a solution for your case?

**Comment by obrazkowv** [comment]
up
